nipype/pipeline/plugins/tests/test_base.py

```diff
@@ -39,3 +40,34 @@ def func(arg1):
 
 wf.run(plugin='MultiProc')
 """
+
+@pytest.mark.regression
+def test_remove_nodes(tmpdir):
+    import os
+    import nipype.interfaces.utility as niu
+    import nipype.pipeline.engine as pe
+
+    wf = pe.Workflow(name='test')
+
+    def func(arg1):
+        try:
+            if arg1 == 2:
+                raise Exception('arg cannot be ' + str(arg1))
+        except:
+            pass
+        return arg1
+
+    funkynode = pe.MapNode(niu.Function(function=func, input_names=['arg1'],
+                                        output_names=['out']),
```

Comment:
This is redundant:

```Python
funkynode = pe.MapNode(niu.Function(function=func),
```

Also, is there something intrinsic about the problem and `MapNode`s? If not, then perhaps just make it a `Node`, to reduce the scope as small as possible.